osu.Game/Overlays/MusicController.cs

```diff
@@ -223,9 +225,12 @@ public bool TogglePause()
         /// </summary>
         /// <param name="onSuccess">Invoked when the operation has been performed successfully.</param>
         /// <param name="allowProtectedTracks">Whether to include <see cref="BeatmapSetInfo.Protected"/> beatmap sets when navigating.</param>
-        public void PreviousTrack(Action<PreviousTrackResult>? onSuccess = null, bool allowProtectedTracks = false) => Schedule(() =>
+        /// <param name="playlist">A playlist to navigate through. Otherwise use the current or list of all beatmap sets.</param>
+        public void PreviousTrack(Action<PreviousTrackResult>? onSuccess = null, bool allowProtectedTracks = false, IEnumerable<Live<BeatmapSetInfo>>? playlist = null) => Schedule(() =>
```

Comment:
The fact that these methods now accept playlists, and that passing one *mutates the internal state* of `MusicController` is dodgy and backwards. Not to mention dangerous when you pass them as `IEnumerable`, which *are generally not safe to enumerate multiple times*.

`MusicController` should be the source of truth regarding what collection is being filtered to, and the order of tracks, not `NowPlayingOverlay`. `NowPlayingOverlay` can _manipulate_ those things via binding to relevant `MusicController` properties or whatever, but it should not be feeding the playlist here like this.